cupy/cuda/compiler.py

```diff
@@ -469,7 +484,8 @@ def _compile_with_cache_cuda(
     if jitify and backend != 'nvrtc':
         raise ValueError('jitify only works with NVRTC')
 
-    env = (arch, options, _get_nvrtc_version(), backend)
+    env_options = options + (_get_arch_for_options(arch, jitify),)
+    env = (arch, env_options, _get_nvrtc_version(), backend)
```

Comment:
How about just adding the option to `env`?

Reply:
do you mean

```
env = (arch, options, _get_nvrtc_version(), backend) + (_get_arch_for_options(arch, jitify),)
```

Reply:
Yes!

```py
env = (arch, options, _get_nvrtc_version(), backend, _get_arch_for_options(arch, jitify))
```